### ISSUE TITLE
[Snackbar] Support custom UIAccessibilityNotification behavior

### ISSUE THREAD

**Issue by romoore** [material-components/material-components-ios#4717]
## Feature Request/Suggestion
Clients should be able to specify whether an `.announcement`, `.layoutChanged`,  or none is posted when a Snackbar appears. The latest internal guidance for Snackbar accessibility is to not change focus automatically.  Since that would be a departure from the current behavior, we should provide an API for clients to migrate smoothly.

See b/112355910 for additional details and internal document references.

---
#### Internal data
- Associated internal bug: [b/112355910](http://b/112355910)

**Comment by romoore** [comment]
Closing this due to inactivity greater than 6 months. Please feel free to reopen if you believe it should be worked on in the near future.
